src/com/amazon/ion/impl/bin/WriteBuffer.java

```diff
@@ -1035,6 +1035,82 @@ public int writeVarUInt(final long value)
         return writeVarUIntSlow(value);
     }
 
+    /** Get the length of varUint for the provided value. */
+    public static int varUIntLength(final long value)
+    {
+        if (value < VAR_UINT_2_OCTET_MIN_VALUE)
+        {
+            return 1;
+        }
+        if (value < VAR_UINT_3_OCTET_MIN_VALUE)
+        {
+            return 2;
+        }
+        if (value < VAR_UINT_4_OCTET_MIN_VALUE)
+        {
+            return 3;
+        }
+        if (value < VAR_UINT_5_OCTET_MIN_VALUE)
+        {
+            return 4;
+        }
+        if (value < VAR_UINT_6_OCTET_MIN_VALUE)
+        {
+            return 5;
+        }
+        if (value < VAR_UINT_7_OCTET_MIN_VALUE)
+        {
+            return 6;
+        }
+        if (value < VAR_UINT_8_OCTET_MIN_VALUE)
+        {
+            return 7;
+        }
+        if (value < VAR_UINT_9_OCTET_MIN_VALUE)
+        {
+            return 8;
+        }
+        throw new IllegalStateException("ion-java failed to support your use case.");
```

Comment:
```suggestion
        return 9;
```
This is actually a legal state, and requires 9 bytes of VarUInt to represent. Since the input is a long, this covers all possible states.